lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1075,12 +1078,28 @@ ONNXModelLoader::loadConstantOfShape(const ONNX_NAMESPACE::NodeProto &op,
     auto TH = in->getPayload().getHandle<int64_t>();
     llvm::ArrayRef<size_t> outputDims = {(const size_t *)TH.begin(),
                                          (const size_t *)TH.end()};
-    ty = G_.getParent()->uniqueType(ElemKind::FloatTy, outputDims);
+    switch (T.getType().getElementType()) {
+    case ElemKind::FloatTy:
+      ty = G_.getParent()->uniqueType(ElemKind::FloatTy, outputDims);
+      SN = G_.createSplat(loadOperatorName(op), ty, T.getHandle().raw(0));
+      break;
+    case ElemKind::Int64ITy:
+      ty = G_.getParent()->uniqueType(ElemKind::Int64ITy, outputDims);
+      SN = G_.createSplat(loadOperatorName(op), ty,
```

Comment:
This will not work.

createSplat takes floating point value of type `float`. Float cannot represent integer number bigger than 16 777 216 exactly (you'll lose precision).

Reply:
You are right. Actually this can happen with int32 as well.  I will put this check -

      int64_t v = T.getHandle<int64_t>().raw(0);
      RETURN_ERR_IF_NOT(v ==(static_cast<int64_t>(static_cast<float>v)),
                        "This ConstantOfShape implementation may cause losses for value " +
                            std::to_string(v)  + " .");

Reply:
> Actually this can happen with int32 as well.

totally, since it's just 16e6 which causes issues. That sounds reasonable short term.